ckan/config/middleware/flask_app.py

```diff
@@ -14,7 +37,18 @@ def make_flask_stack(conf, **app_conf):
     """ This has to pass the flask app through all the same middleware that
     Pylons used """
 
+    root = os.path.dirname(
+        os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```

Comment:
We lose support for deploying as an archive or other importers doing this don't we? Although I bet it's already broken for similar things in other places. We should use the helpers in `pkg_resources` to get the true root.